File: backend/app/models/categoria.py
from typing import List, Optional
from datetime import datetime
from sqlmodel import SQLModel, Field, Relationship


class Categoria(SQLModel, table=True):
    id: Optional[int] = Field(default=None, primary_key=True)
    parent_id: Optional[int] = Field(default=None, foreign_key="categoria.id")
    nombre: str = Field(max_length=100)
    descripcion: Optional[str] = None
    imagen_url: Optional[str] = None
    created_at: datetime = Field(default_factory=datetime.utcnow)
    updated_at: datetime = Field(default_factory=datetime.utcnow)
    deleted_at: Optional[datetime] = Field(default=None)

    productos: List["ProductoCategoria"] = Relationship(back_populates="categoria")

File: frontend/src/types/categoria.ts
export interface Categoria {
  id: number
  nombre: string
  descripcion?: string
  imagen_url?: string
  parent_id?: number
  created_at: string
  updated_at: string
}

export interface CategoriaCreate {
  nombre: string
  descripcion?: string
  imagen_url?: string
  parent_id?: number
}